workflow/controller/artifact_gc.go

```diff
@@ -179,10 +179,7 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 			woc.log.Errorf("Was unable to obtain node for %s", artifactSearchResult.NodeID)
 			return fmt.Errorf("can't process Artifact GC Strategy %s: node ID %q not found in Status??", strategy, artifactSearchResult.NodeID)
 		}
-		templateName := node.TemplateName
-		if templateName == "" && node.GetTemplateRef() != nil {
-			templateName = node.GetTemplateRef().Template
-		}
+		templateName := util.GetTemplateFromNode(*node)
```

Comment:
I believe `func (n *NodeStatus) GetTemplateRef() *TemplateRef {` in `workflow_types.go` is no longer needed and can be removed now

Reply:
oh maybe not... `TemplateReferenceHolder` appears to be a supertype... although it's not explicitly used that way... it looks like it's being used as an implicit structural type... that's confusing... should probably be refactored to be a properly inherited nominal type

Reply:
and `GetTemplateName` is technically correct since it's supposed to return the raw `TemplateName`... yea that type is really confusing... there's almost certainly some years old tech debt with the structural types 😕